### ISSUE TITLE
[Core] Update LRO delay strategies to use new public DelayStrategy type

### ISSUE THREAD

**Issue by JoshLove-msft** [Azure/azure-sdk-for-net#35437]
Once the new version of Core is shipped with the DelayStrategy type, we will need to:
- delete `DelayStrategyInternal` and update all subclasses to inherit from DelayStrategy.  
- remove any references to DelayStrategyInternal from csproj files
- All existings test strategy implementations will need to be updated as well. 
- `ExponentialDelayStrategyInternal` should be renamed to `ExponentialDelayStrategy`

Basically, we need to fix all the breaks once DelayStrategyInternal is deleted by instead referencing DelayStrategy.

All of this is actually already done in https://github.com/Azure/azure-sdk-for-net/pull/33040, but we needed to pull off the DelayStrategy piece and ship it separately to avoid requiring project references to Core on all mgmt libraries.
